libs/designer/src/lib/ui/mcp/parameters/parameterfield.tsx

```diff
@@ -0,0 +1,233 @@
+    operationInfo: state.operations.operationInfo[operationId],
+    reference: state.connection.connectionReferences[state.connection.connectionsMapping[operationId] ?? ''],
+    nodeInputs: state.operations.inputParameters[operationId],
+    dependencies: state.operations.dependencies[operationId],
+  }));
+
+  const INTL_TEXT = {
+    removeParamText: intl.formatMessage({
+      id: '5E66mK',
+      defaultMessage: 'Remove parameter',
+      description: 'Tooltip for remove parameter button',
+    }),
+    modelRadioText: intl.formatMessage({
+      id: 'LNA+DZ',
+      defaultMessage: 'Model',
+      description: 'Label for parameter to use model input type',
+    }),
+    userRadioText: intl.formatMessage({
+      id: 'gOWYv9',
+      defaultMessage: 'User',
+      description: 'Label for parameter to use user input type',
+    }),
+    inputLabelText: intl.formatMessage({
+      id: 'gwYMqA',
+      defaultMessage: 'Input',
+      description: 'Label for input field in parameter editor radio button section',
+    }),
+  };
+
+  const onHandleRemoveError = useCallback(() => {
+    removeParameterError(parameter.id as string);
+  }, [parameter.id, removeParameterError]);
+
+  const isLargeParameter = useMemo(() => {
+    const editor = parameter.editor?.toLowerCase();
+    return (
+      editor === constants.EDITOR.ARRAY ||
+      editor === constants.EDITOR.DICTIONARY ||
+      editor === constants.EDITOR.HTML ||
+      editor === constants.EDITOR.TABLE
+    );
+  }, [parameter.editor]);
+
+  const onParameterValueChange = useCallback(
+    (newState: ChangeState) => {
+      const { value: newValueSegments, viewModel } = newState;
+      const propertiesToUpdate = { value: newValueSegments, preservedValue: undefined } as Partial<ParameterInfo>;
+
+      if (viewModel !== undefined) {
+        propertiesToUpdate.editorViewModel = viewModel;
+      }
+
+      const nodeId = operationId;
+      const parameterId = parameter.id as string;
+
+      if (parameterHasValue(propertiesToUpdate as ParameterInfo)) {
+        onHandleRemoveError();
+      }
+
+      const updateParameterAndDependencyPayload: UpdateParameterAndDependenciesPayload = {
+        nodeId,
+        groupId,
+        parameterId,
+        properties: propertiesToUpdate,
+        isTrigger: false,
+        operationInfo,
+        connectionReference: reference,
+        nodeInputs,
+        dependencies,
+        updateTokenMetadata: false,
+      };
+
+      dispatch(updateParameterAndDependencies(updateParameterAndDependencyPayload));
+
+      const updateParametersPayload: UpdateParametersPayload = {
+        nodeId,
+        parameters: [
+          {
+            groupId,
+            parameterId,
+            propertiesToUpdate: { value: newValueSegments },
+          },
+        ],
+        isUserAction: true,
+      };
+      dispatch(updateNodeParameters(updateParametersPayload));
```

Comment:
We don tneed to do this, this is already happening in the above dispatch function